src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -106,17 +112,35 @@ LinalgBackendBase* infer_backend(const Container<T>& a, const Container<T>& b)
  * and a shallow-copy of SGVector with vector on CPU if GPU backend not available
  */
 template <typename T>
-SGVector<T> to_gpu(const SGVector<T>& vector)
+void to_gpu(SGVector<T>& a, SGVector<T>& b)
 {
```

Comment:
@karlnapf @vigsterkr Regardless of the technical details, I am reshaping `a = to_gpu(a)` to `to_gpu(a, a)` and/or `to_gpu(a)` as the copy assignment `a = xxx` is not atomic.